test/integration/job/job_test.go

```diff
@@ -1716,7 +1717,10 @@ func TestJobPodReplacementPolicy(t *testing.T) {
 				CompletionMode:       &nonIndexedCompletion,
 				PodReplacementPolicy: podReplacementPolicy(batchv1.Failed),
 			},
-			wantTerminating: pointer.Int32(podCount),
+			failPodsInsteadOfDeletion: true,
+			wantActive:                int(podCount),
+			wantFailed:                int(podCount),
+			wantTerminating:           pointer.Int32(0),
```

Comment:
I think you should just drop `wantTerminating`.

Reply:
But I expect it to be `*int32(0)`, otherwise it will be `null`